src/client/client.ts

```diff
@@ -273,6 +280,19 @@ export class Client {
       authorization: `Bearer ${config.token}`,
     });
     this.token = config.token;
+    //@ts-expect-error caused by undici and bunjs type overlap
+    this.headers = prefixHeaders(new Headers(config.headers));
+  }
+
+  private wrapWithGlobalHeaders(headers: Headers) {
+    const finalHeaders = new Headers(this.headers);
+
+    // eslint-disable-next-line unicorn/no-array-for-each
+    headers.forEach((value, key) => {
+      finalHeaders.set(key, value);
+    });
+
+    return finalHeaders;
   }
```

Comment:
Why do we have this function? 
```suggestion
  private wrapWithGlobalHeaders(headers: Headers) {
    const finalHeaders = new Headers(this.headers);

    // eslint-disable-next-line unicorn/no-array-for-each
    headers.forEach((key, value) => {
      finalHeaders.set(key, value);
    });

    return finalHeaders;
  }
```

Why do we even have this function? Are we trying to reverse the headers' key value? The user should already give them as

```
    const clientWithHeaders = new Client({
      token: process.env.QSTASH_TOKEN!,
      headers: {
        "test-header": "value",
        "test-header-2": "value-2",
      },
    });
```

Reply:
It is [reversed in the api for no reason](https://developer.mozilla.org/en-US/docs/Web/API/Headers/forEach). It seemed weird to me too